Append -funroll-loops flag to any existing CFLAGS.
Remove -frename-registers which no version of clang understands.
Use CPPFLAGS when compiling.
Handle DESTDIR variable to properly install stuff.
--- Makefile.orig
+++ Makefile
@@ -1,6 +1,6 @@
 PREFIX    ?= /usr/local
 CC         = gcc
-CFLAGS    ?= -O2 -g -funroll-loops -frename-registers
+CFLAGS    += -funroll-loops
 CFLAGS    += -D_LARGEFILE_SOURCE -DLARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -D_XOPEN_SOURCE=500 -D_BSD_SOURCE -D_REENTRANT
 CFLAGS    += -Wall
 OBJECTS    = main.o
@@ -14,14 +14,14 @@ M2VRequantiser : $(OBJECTS)
 	$(CC) $(CFLAGS) $(OBJECTS) $(LDFLAGS) $(LOADLIBES) -o M2VRequantiser
 
 %.o : %.c
-	$(CC) $(CFLAGS) -c $<
+	$(CC) $(CPPFLAGS) $(CFLAGS) -c $<
 
 install:	all
-	$(INSTALL) -d $(PREFIX)/bin
-	$(INSTALL) -m 755 M2VRequantiser $(PREFIX)/bin
+	$(INSTALL) -d $(DESTDIR)$(PREFIX)/bin
+	$(INSTALL) -m 755 M2VRequantiser $(DESTDIR)$(PREFIX)/bin
 	
 uninstall:
-	rm -f $(PREFIX)/bin/M2VRequantiser
+	rm -f $(DESTDIR)$(PREFIX)/bin/M2VRequantiser
 
 clean:
 	rm -f M2VRequantiser *.o *~ core
